lib/LocalModel.js

```diff
@@ -468,6 +495,171 @@ LocalModel.prototype.predictProbability = function (inputData, missingStrategy,
+                       "confidence": 0.0});
+        }
+      }
+      if (cb) {
+        return cb(null, output);
+      }
+      return output;
+    }
+
+  if (this.ready) {
+    if (cb) {
+      this.predict(inputData, missingStrategy, false, false,
+                   syncPredictConfidence);
+    } else {
+      prediction = syncPredictConfidence(null,
+        this.predict(inputData, missingStrategy, false, false, undefined));
+      return prediction;
+    }
+  } else {
+    this.on('ready', function (self) {
+      return self.predictConfidence(inputData, missingStrategy, cb);
+    });
+    return;
+  }
+};
+
+LocalModel.prototype.predictOperating = function (
+  inputData, missingStrategy, operatingPoint, cb) {
+  /**
+   * For classification models, predicts using the operation point
+   *    defined by a positive class and some threshold and
+   *    based on input values.  The input
+   *    fields must be a dictionary keyed by field name or field ID.
+   *
+   *    For regressions, the output is a single element list
+   *    containing the prediction.
+   *
+   * @param {object} inputData Input data to predict
+   * @param {{0|1}]} missingStrategy Strategy for missing values
+   * @param {object} operatingPoint Operating point definition. The object
+   *                                should contain a positiveClass and
+   *                                either a probabilityThreshold,
+   *                                confidenceThreshold or a
+   *                                kThreshold
+   * @param {function} cb Callback
+   */
+
+  var positiveClass, kind, threshold, predictMethod, issuePrediction,
+    prediction, predictions, position, index, sortFn, self=this;
+
+  if ((typeof missingStrategy === 'function') && (typeof cb === 'undefined')) {
+    cb = missingStrategy;
+    missingStrategy = constants.LAST_PREDICTION;
+  }
+  if ((typeof operatingPoint === 'function') && (typeof cb === 'undefined')) {
+    cb = operatingPoint;
+  }
+
+  if (this.ready) {
+    missingStrategy = missingStrategy || constants.LAST_PREDICTION;
+    utils.checkOperatingPoint(operatingPoint,
+                              MODEL_OPERATING_KINDS,
+                              this.classNames);
+    positiveClass = operatingPoint.positiveClass;
+    kind = operatingPoint.kind;
+    predictMethod = "predict" + kind.charAt(0).toUpperCase() + kind.slice(1)
+    threshold = operatingPoint.threshold;
+    issuePrediction = function(error, predictions) {
+      position = self.classNames.indexOf(positiveClass);
+      if (predictions[position][kind] < threshold) {
+        // if the threshold is not met, the alternative class with
+        // highest probability or confidence is returned
+        sortFn = function (a, b) {
+          return b[kind] - a[kind];}
+        predictions.sort(sortFn);
+        prediction = (predictions[0]["prediction"] == positiveClass) ? predictions[1] : predictions[0];
+      } else {
```

Comment:
the logic above for applying an operating point looks very similar to the code used for local ensembles, and close to that of LRs.  perhaps it could be refactored into a function?